proposals/42-human-readable-mutable-names.md

```diff
@@ -0,0 +1,235 @@
+
+**TLDR:** **We need to fix and future-proof the way mutable names work in our stack:**
+- Make IPNS useful on its own, and inside of DNSLink records.
+    - DNS TXT record should be set only once, updates should be published over IPNS.
+- Future-proof the way we do human readable names by
+    - leveraging DNS protocol (not the DNS network run by ICANN) for interop with existing and future user agents and naming systems
+    - allowing flexible configuration to improve security (DoH) and interop (DNSLink)
+
+
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Browser is the  main distribution channel for dapps.
+- Dapps are simply a subset of all static websites loaded from IPFS.
+- Most of websites hosted on IPFS have DNSLink set up for human-readable name.
+- We have mainstream browsers like Opera and [Brave](https://brave.com/ipfs-support/) ship support for `ipfs://` and `ipns://` URIs. 
+    - Human readable names like `ipns://en.wikipedia-on-ipfs.org` work thanks to DNSLink
+    - This removes huge UX friction for dapp developers. 
+    - Vendors like Brave are committed to [surfacing DNSLink supportin the UI](https://github.com/brave/brave-browser/issues/13609) which will improve onboarding even further.
+- Alternatives to [ICANN](https://en.wikipedia.org/wiki/ICANN)  exist, and we want them to work with our stack to decrease dependency on a single organization.
+    - [ENS](https://ens.domains) provides custom TLDs (`.eth`) and leverage DNSLink for interop with IPFS and delegated lookups with clients that are unable to run own blockchain resolver.
+        - Updating `ipfs-ns` `contenthash` is a chore and costs extra (gas etc). 
+        - Using `ipns-ns` means setting `contenthash` only once, introduces cost savings and simplifies publishing. 
+    - We see multiple actors in the space providing either own TLDs or attempting to replace ICANN as the top-level authority: https://unstoppabledomains.com , http://opennic.org, https://handshake.org etc.
+    - We want to enable innovation in the decentralized naming space. This means removing ourselves as gatekeepers of what a valid domain name is.
+    - When embedded in user agent (Brave) we want to follow user choices regarding DNS resolution.
+    
+
+
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+- Improving human-readable naming removes huge DX/UX friction for dapp users and developers. 
+    - Messaging collapses to "Install Brave or Opera and open ipns://example.com"
+    - They use IPNS for publishing updates without ever touching DNS/ENS etc beyond initial setup. Most likely save money on gas and/or admin time.
+    - One-liner for setting up a custom DNSLink resolver improves onboarding for naming systems not supported by default and enables vendors like Brave to provide UI for changing resolution settings in more user-friendly manner.
```

Comment:
> enables vendors like Brave to provide UI for changing resolution settings in more user-friendly manner

I'm not quite following this.  Can you add a bit more detail?

Reply:
I meant that Brave may provide UI in their settings for changing default DNS over HTTPS resolver endpoint, or one for resolving ENS.